Automattic/jetpack
# Update/purchases state management

## Proposed changes:

* Convert the purchases data query from redux to Tanstack useQuery
* Create pattern for how future data queries will be made
* Add a React Context version of Global Notices to eventually remove it from redux (after all redux use cases are removed one by one by data query)
* Fix a couple of console errors having to do with types

This should lay the groundwork for replacing all redux data types in My Jetpack with Tanstack's useQuery. The Global notices are not a "data query" so tanstack didn't seem like the right way (nor do I think it is possible) to replace that part of the redux store. React context seemed, to me, the right way to go in that case

### Other information:


## Jetpack product discussion

p1HpG7-rcF-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. On either the Beta plugin or your local dev environment, make sure you are Sandboxing the Jetpack API (not completely necessary if you are using your local dev environment, but the testing instructions will be editing a WPCOM API since that works for both the beta plugin and local environment)
3. Make a purchase (VaultPress Backup, for example), and go to `/wp-admin/admin.php?page=my-jetpack`
4. Scroll down and make sure the purchase loads correctly and shows up under Your Plan
![image](https://github.com/Automattic/jetpack/assets/65001528/9ae49ca1-b24f-4c61-b899-c9c3194a4418)
5. Next, in WPCOM, go to this class: fbhepr%2Skers%2Sjcpbz%2Schoyvp.ncv%2Serfg%2Sjcpbz%2Qwfba%2Qraqcbvagf%2Spynff.jcpbz%2Qfgber%2Qncv%2Qraqcbvagf.cuc%3Se%3Q1prs6727%23780-og
and add `return new WP_Error();` at the beginning of the callback function
6. Reload My Jetpack and make sure the global notice shows up as expected
![image](https://github.com/Automattic/jetpack/assets/65001528/ec2f9316-5adc-42c8-aac0-8ed4d4cc334b)
7. Go through the code carefully and see if you can come up with any ways it could be even more streamlined. This will be how we do data queries going forward in My Jetpack so we should get it as good as we can right off the bat
